ivy_tests/test_ivy/helpers/assertions.py

```diff
@@ -148,6 +152,12 @@ def value_test(
             )
     else:
         for ret_np, ret_np_from_gt in zip(ret_np_flat, ret_np_from_gt_flat):
+            if specific_tolerance_dict is not None:
```

Comment:
above in the `if` statement we're only checking if `rolt` is given or not then we're getting `rtol` from global `TOLERANCE_DICT`. This should be modified so that it would first check if `specific_tolerance_dict` exists if it doesn't then it would check if `rtol` exists, if that doesn't then it would get it from the global `TOLERANCE_DICT`.

the priority should be in the following order:
`specific_tolerance_dict`
`rtol`
`TOLERANCE_DICT`